src/dirent_lookup.h

```diff
@@ -91,93 +147,39 @@ DirentLookup<Impl>::getNamespaceRangeEnd(char ns) const
   return getNamespaceRangeBegin(ns+1);
 }
 
-template<class Impl>
-typename DirentLookup<Impl>::DirentRange
-DirentLookup<Impl>::getDirentRange(char ns, const std::string& /*url*/) const
-{
-  DirentRange r;
-  r.begin = getNamespaceRangeBegin(ns);
-  r.end   = getNamespaceRangeEnd(ns);
-  return r;
-}
-
 template<typename Impl>
 typename DirentLookup<Impl>::Result
 DirentLookup<Impl>::find(char ns, const std::string& url)
 {
-  const DirentRange r = getDirentRange(ns, url);
+  const auto r = lookupGrid.getRange(ns + url);
   article_index_type l(r.begin);
   article_index_type u(r.end);
 
   if (l == u)
-  {
-    return std::pair<bool, article_index_t>(false, article_index_t(0));
-  }
+    return {false, article_index_t(0)};
 
-  unsigned itcount = 0;
-  while (u - l > 1)
+  while (true)
   {
-    ++itcount;
     article_index_type p = l + (u - l) / 2;
-    auto d = impl.getDirent(article_index_t(p));
+    const auto d = impl->getDirent(article_index_t(p));
 
-    int c = ns < d->getNamespace() ? -1
-          : ns > d->getNamespace() ? 1
-          : url.compare(d->getUrl());
+    const int c = ns < d->getNamespace() ? -1
+                : ns > d->getNamespace() ? 1
+                : url.compare(d->getUrl());
```

Comment:
Here is something I potentially wanted to change but never does :
As the range (l-r) is (was) namespace boundary it should be useless to check the namespace. We could simply compare the url.

This PR remove the fact that the range contains only one namespace so we still need to keep the namespace compare. 
Maybe we can go with a more common idea and keep track of the prefix length we are sure to have (if `l` and `r`'s urls start with the same N char, we know that the middle dirent will do the same, we can start the url comparison from array N)
But I'm not sure that keeping track of the prefix is more efficient than a simple comparison.

Reply:
I had the same idea for compressing the key storage even further by encoding the shared prefixes in a special way, which would also automatically eliminate the shared prefix from the string comparison. But I decided not to include that optimization in this PR.